integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/l0promotion/DistroXOSUpgradeTests.java

```diff
@@ -0,0 +1,144 @@
+package com.sequenceiq.it.cloudbreak.testcase.e2e.l0promotion;
+
+import static com.sequenceiq.it.cloudbreak.context.RunningParameter.key;
+
+import javax.inject.Inject;
+
+import org.testng.annotations.Test;
+
+import com.sequenceiq.distrox.api.v1.distrox.model.upgrade.DistroXUpgradeReplaceVms;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.it.cloudbreak.client.DistroXTestClient;
+import com.sequenceiq.it.cloudbreak.client.EnvironmentTestClient;
+import com.sequenceiq.it.cloudbreak.client.FreeIpaTestClient;
+import com.sequenceiq.it.cloudbreak.client.SdxTestClient;
+import com.sequenceiq.it.cloudbreak.cloud.v4.CommonClusterManagerProperties;
+import com.sequenceiq.it.cloudbreak.context.Description;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.distrox.DistroXTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.cluster.DistroXUpgradeTestDto;
+import com.sequenceiq.it.cloudbreak.dto.freeipa.FreeIpaTestDto;
+import com.sequenceiq.it.cloudbreak.dto.sdx.SdxTestDto;
+import com.sequenceiq.it.cloudbreak.dto.sdx.SdxUpgradeTestDto;
+import com.sequenceiq.it.cloudbreak.testcase.e2e.AbstractE2ETest;
+import com.sequenceiq.it.cloudbreak.util.spot.UseSpotInstances;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+import com.sequenceiq.sdx.api.model.SdxUpgradeReplaceVms;
+
+public class DistroXOSUpgradeTests extends AbstractE2ETest {
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private DistroXTestClient distroXTestClient;
+
+    @Inject
+    private CommonClusterManagerProperties commonClusterManagerProperties;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private EnvironmentTestClient environmentTestClient;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        testContext.getCloudProvider().getCloudFunctionality().cloudStorageInitialize();
+        createDefaultUser(testContext);
+        initializeDefaultBlueprints(testContext);
+        createDefaultCredential(testContext);
+        createEnvironmentWithFreeIpa(testContext);
+    }
+
+    @Test(dataProvider = TEST_CONTEXT, description = "We need to wait the OPSAPS-62124 to be resolved")
```

Comment:
Why  did we add @UseSpotInstances?

if this is in the l0promotion won't this be automatically triggered by the L0 tests? or do we have an explicit list there? For this reason I would add a disabled test notation just to be safe that this won't be included accidentally.

Reply:
Right now promotion tests are defined suite-by-suite with own YAML file:
- [cloudbreak-ansible-playbooks/jenkins/jenkins_jobs/mow-api-e2e-aws.xml](https://github.com/hortonworks/cloudbreak-ansible-playbooks/blob/master/jenkins/jenkins_jobs/mow-api-e2e-aws.xml#L583)
- [cloudbreak-ansible-playbooks/jenkins/jenkins_jobs/mow-api-e2e-azure.xml](https://github.com/hortonworks/cloudbreak-ansible-playbooks/blob/master/jenkins/jenkins_jobs/mow-api-e2e-azure.xml#L521)

However we can rest assured by the mentioned [@Ignore TestNG annotation](https://testng.org/doc/documentation-main.html#ignore-tests). So I've introduced this.